src/hotspot/cpu/x86/x86.ad

```diff
@@ -9479,6 +9487,49 @@ instruct mask_not_imm(kReg dst, kReg src, immI_M1 cnt) %{
   %}
   ins_pipe( pipe_slow );
 %}
+
+instruct long_to_maskLE8_avx(vec dst, rRegL src, rRegL rtmp1, rRegL rtmp2, vec xtmp) %{
+  predicate(n->bottom_type()->isa_vectmask() == NULL && Matcher::vector_length(n) <= 8);
+  match(Set dst (VectorLongToMask src));
+  effect(TEMP dst, TEMP rtmp1, TEMP rtmp2, TEMP xtmp);
+  format %{ "long_to_mask_avx $dst, $src\t! using $rtmp1, $rtmp2, $xtmp as TEMP" %}
+  ins_encode %{
+    int mask_len = Matcher::vector_length(this);
+    int vec_enc  = vector_length_encoding(mask_len*8);
+    __ vector_long_to_maskvec($dst$$XMMRegister, $src$$Register, $rtmp1$$Register,
+                              $rtmp2$$Register, xnoreg, mask_len, vec_enc);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+
+instruct long_to_maskGT8_avx(vec dst, rRegL src, rRegL rtmp1, rRegL rtmp2, vec xtmp1, vec xtmp2, rFlagsReg cr) %{
+  predicate(n->bottom_type()->isa_vectmask() == NULL && Matcher::vector_length(n) > 8);
+  match(Set dst (VectorLongToMask src));
+  effect(TEMP dst, TEMP rtmp1, TEMP rtmp2, TEMP xtmp1, TEMP xtmp2, KILL cr);
+  format %{ "long_to_mask_avx $dst, $src\t! using $rtmp1, $rtmp2, $xtmp1, $xtmp2  as TEMP" %}
+  ins_encode %{
+    int mask_len = Matcher::vector_length(this);
+    assert(mask_len <= 32, "invalid mask length");
+    int vec_enc  = vector_length_encoding(mask_len*8);
+    __ vector_long_to_maskvec($dst$$XMMRegister, $src$$Register, $rtmp1$$Register,
+                              $rtmp2$$Register, $xtmp1$$XMMRegister, mask_len, vec_enc);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct long_to_mask_evex(kReg dst, rRegL src, rRegL rtmp) %{
+  predicate(n->bottom_type()->isa_vectmask());
+  match(Set dst (VectorLongToMask src));
+  effect(TEMP rtmp);
+  format %{ "long_to_mask_evex $dst, $src\t! using $rtmp as TEMP" %}
+  ins_encode %{
+    int mask_len = Matcher::vector_length(this);
+    __ movq($rtmp$$Register, $src$$Register);
+    __ kmov($dst$$KRegister, $rtmp$$Register);
```

Comment:
why extra move to rtmp here? Cannot we generate directly kmov(dst, src)?

Reply:
DONE